tools/getrealdeps.sh

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env bash
+# getrealdeps for Chromebrew
+# Author: Satadru Pramanik (satmandu) satadru at gmail dot com
+# set -x
+pkg="${1}"
+
+RED='\e[1;91m';    # Use Light Red for errors.
+YELLOW='\e[1;33m'; # Use Yellow for informational messages.
+GREEN='\e[1;32m';  # Use Green for success messages.
+BLUE='\e[1;34m';   # Use Blue for intrafunction messages.
+GRAY='\e[0;37m';   # Use Gray for program output.
+MAGENTA='\e[1;35m';
+RESET='\e[0m'
+
+# Simplify colors and print errors to stderr (2).
+echo_error() { echo -e "${RED}${*}${RESET}" >&2; }
+echo_info() { echo -e "${YELLOW}${*}${RESET}" >&1; }
+echo_success() { echo -e "${GREEN}${*}${RESET}" >&1; }
+echo_intra() { echo -e "${BLUE}${*}${RESET}" >&1; }
+echo_out() { echo -e "${GRAY}${*}${RESET}" >&1; }
+echo_other() { echo -e "${MAGENTA}${*}${RESET}" >&2; }
+
+# Install grep if a functional local copy does not exist.
+if grep --version &> /dev/null; then
+  GREP=grep
+else
+  crew install grep
+  GREP=/usr/local/bin/grep
+fi
+
+: "${CREW_LIB_PREFIX:=$(crew const | $GREP CREW_LIB_PREFIX | awk -F = '{print $2}')}"
+
+# Which packages have a needed library in CREW_LIB_PREFIX
+# This is a subset of what crew whatprovides gives
+whatprovidesfxn() {
+  pkgdepslcl="${1}"
+  filelcl=$($GREP --exclude "${pkg}.filelist" "$pkgdepslcl" /usr/local/etc/crew/meta/*.filelist | $GREP "$CREW_LIB_PREFIX")
+  packagelcl=$(echo "$filelcl" | \
+  sed 's/.filelist.*//g' | sed 's:.*/::' | awk '!x[$0]++' | sed s/://g)
+  echo "$packagelcl"
+}
+
+# What files does a package provide
+crewfilesfxn() {
+ pkgname="${1}"
+ files=$(< /usr/local/etc/crew/meta/"${pkgname}".filelist)
+ echo "$files"
+}
+
+pkgfiles=$(crewfilesfxn "${pkg}")
+# Use readelf to determine library dependencies, as
+# this doesn't almost run a program like using ldd would.
+rm -rf /tmp/deps/"${pkg}"
+pkgdeps=$(unset lines ; for i in $pkgfiles ; \
```

Comment:
Add to variable inside `for` loop instead of capturing the whole block?
```shell
pkgdeps=''
for i in "${pkgfiles}"; do
  [[ "$(head -c 4 "${i}")" == $'\x7FELF' ]] || continue
  [[ $i = *_tkinter* ]] && continue # Carveout for Python3
  mkdir -p /tmp/deps/"${pkg}"/

  basename="${a##*/}" # equivalent to basename
  pkgdeps+="$(readelf -d "$i" 2>/dev/null | $GREP NEEDED | awk '{print $5}' | tr -d '[]' | awk '!x[$0]++' | tee "/tmp/deps/${pkg}/${basename}")" # use `tr` to delete words + wrap with " to preserve newlines
done

pkgdeps="$(awk 'NF' <<< "${pkgdeps}" | awk '!x[$0]++'))"
```

Reply:
The for loop doesn't work, sadly.

Try it with just `./getrealdeps.sh libcap`

```
chronos@cheekon-x86_64 /usr/local/lib/crew/packages $ ../tools/getrealdeps.sh libcap
head: cannot open '/usr/local/include/sys/capability.h'$'\n''/usr/local/include/sys/psx_syscall.h'$'\n''/usr/local/lib64/pkgconfig/libcap.pc'$'\n''/usr/local/lib64/pkgconfig/libpsx.pc'$'\n''/usr/local/lib64/libcap.a'$'\n''/usr/local/lib64/libpsx.a'$'\n''/usr/local/lib64/libcap.so.2.66'$'\n''/usr/local/lib64/libpsx.so.2.66'$'\n''/usr/local/sbin/getpcaps'$'\n''/usr/local/sbin/getcap'$'\n''/usr/local/sbin/setcap'$'\n''/usr/local/sbin/capsh'$'\n''/usr/local/share/man/man1/capsh.1.gz'$'\n''/usr/local/share/man/man3/cap_setgroups.3.gz'$'\n''/usr/local/share/man/man3/cap_set_mode.3.gz'$'\n''/usr/local/share/man/man3/capsetp.3.gz'$'\n''/usr/local/share/man/man3/cap_set_proc.3.gz'$'\n''/usr/local/share/man/man3/cap_set_secbits.3.gz'$'\n''/usr/local/share/man/man3/cap_setuid.3.gz'$'\n''/usr/local/share/man/man3/cap_size.3.gz'$'\n''/usr/local/share/man/man3/cap_to_name.3.gz'$'\n''/usr/local/share/man/man3/cap_to_text.3.gz'$'\n''/usr/local/share/man/man3/libcap.3.gz'$'\n''/usr/local/share/man/man3/libpsx.3.gz'$'\n''/usr/local/share/man/man3/psx_set_sensitivity.3.gz'$'\n''/usr/local/share/man/man3/psx_syscall.3.gz'$'\n''/usr/local/share/man/man3/psx_syscall3.3.gz'$'\n''/usr/local/share/man/man3/psx_syscall6.3.gz'$'\n''/usr/local/share/man/man3/cap_max_bits.3.gz'$'\n''/usr/local/share/man/man3/cap_mode.3.gz'$'\n''/usr/local/share/man/man3/cap_mode_name.3.gz'$'\n''/usr/local/share/man/man3/cap_set_fd.3.gz'$'\n''/usr/local/share/man/man3/cap_set_flag.3.gz'$'\n''/usr/local/share/man/man3/cap_init.3.gz'$'\n''/usr/local/share/man/man3/cap_launcher_callback.3.gz'$'\n''/usr/local/share/man/man3/cap_launcher_setgroups.3.gz'$'\n''/usr/local/share/man/man3/cap_launcher_set_mode.3.gz'$'\n''/usr/local/share/man/man3/cap_new_launcher.3.gz'$'\n''/usr/local/share/man/man3/cap_clear.3.gz'$'\n''/usr/local/share/man/man3/cap_clear_flag.3.gz'$'\n''/usr/local/share/man/man3/cap_compare.3.gz'$'\n''/usr/local/share/man/man3/cap_copy_ext.3.gz'$'\n''/usr/local/share/man/man3/cap_copy_int.3.gz'$'\n''/usr/local/share/man/man3/cap_drop_bound.3.gz'$'\n''/usr/local/share/man/man3/cap_dup.3.gz'$'\n''/usr/local/share/man/man3/cap_fill.3.gz'$'\n''/usr/local/share/man/man3/cap_fill_flag.3.gz'$'\n''/usr/local/share/man/man3/cap_free.3.gz'$'\n''/usr/local/share/man/man3/cap_from_name.3.gz'$'\n''/usr/local/share/man/man3/cap_from_text.3.gz'$'\n''/usr/local/share/man/man3/cap_func_launcher.3.gz'$'\n''/usr/local/share/man/man3/cap_get_bound.3.gz'$'\n''/usr/local/share/man/man3/cap_get_fd.3.gz'$'\n''/usr/local/share/man/man3/cap_get_file.3.gz'$'\n''/usr/local/share/man/man3/cap_get_flag.3.gz'$'\n''/usr/local/share/man/man3/cap_get_mode.3.gz'$'\n''/usr/local/share/man/man3/capgetp.3.gz'$'\n''/usr/local/share/man/man3/cap_get_pid.3.gz'$'\n''/usr/local/share/man/man3/cap_get_proc.3.gz'$'\n''/usr/local/share/man/man3/cap_get_secbits.3.gz'$'\n''/usr/local/share/man/man3/cap_iab.3.gz'$'\n''/usr/local/share/man/man3/cap_iab_compare.3.gz'$'\n''/usr/local/share/man/man3/cap_iab_dup.3.gz'$'\n''/usr/local/share/man/man3/cap_iab_fill.3.gz'$'\n''/usr/local/share/man/man3/cap_iab_from_text.3.gz'$'\n''/usr/local/share/man/man3/cap_iab_get_pid.3.gz'$'\n''/usr/local/share/man/man3/cap_iab_get_proc.3.gz'$'\n''/usr/local/share/man/man3/cap_iab_get_vector.3.gz'$'\n''/usr/local/share/man/man3/cap_iab_init.3.gz'$'\n''/usr/local/share/man/man3/cap_iab_set_proc.3.gz'$'\n''/usr/local/share/man/man3/cap_iab_set_vector.3.gz'$'\n''/usr/local/share/man/man3/cap_iab_to_text.3.gz'$'\n''/usr/local/share/man/man3/cap_launch.3.gz'$'\n''/usr/local/share/man/man3/cap_launcher_set_chroot.3.gz'$'\n''/usr/local/share/man/man3/cap_launcher_set_iab.3.gz'$'\n''/usr/local/share/man/man3/cap_launcher_setuid.3.gz'$'\n''/usr/local/share/man/man3/cap_set_file.3.gz'$'\n''/usr/local/share/man/man8/captree.8.gz'$'\n''/usr/local/share/man/man8/getcap.8.gz'$'\n''/usr/local/share/man/man8/getpcaps.8.gz'$'\n''/usr/local/share/man/man8/setcap.8.gz'$'\n''/usr/local/lib64/libcap.so.2'$'\n''/usr/local/lib64/libcap.so'$'\n''/usr/local/lib64/libpsx.so.2'$'\n''/usr/local/lib64/libpsx.so' for reading: No such file or directory

Package libcap has these runtime dependencies:



chronos@cheekon-x86_64 /usr/local/lib/crew/packages $
```

Reply:
working version:
```
chronos@cheekon-x86_64 /usr/local/lib/crew/packages $ ../tools/getrealdeps.sh libcap

Package libcap has these runtime dependencies:

  depends_on 'glibc' # R


chronos@cheekon-x86_64 /usr/local/lib/crew/packages $ 
```